crates/sequencer/src/state.rs

```diff
@@ -22,13 +24,13 @@ use crate::commit::Committer;
 /// which is faster than the default hash function. Think about changing
 /// if the test sequencer is used for tests outside of ef-tests.
 /// See [rustc-hash](https://crates.io/crates/rustc-hash) for more information.
-#[derive(Default)]
+#[derive(Clone, Default, Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub struct State {
-    classes: FxHashMap<ClassHash, ContractClass>,
-    compiled_class_hashes: FxHashMap<ClassHash, CompiledClassHash>,
-    contracts: FxHashMap<ContractAddress, ClassHash>,
-    storage: FxHashMap<ContractStorageKey, StarkFelt>,
-    nonces: FxHashMap<ContractAddress, Nonce>,
+    pub(crate) classes: FxHashMap<ClassHash, ContractClass>,
```

Comment:
why make this pub crate?

Reply:
pretty sure you need to make it at least pub(crate) if you want to serialize

Reply:
ah i see